src/bundle/ui-dev/src/modules/multi-file-upload/components/upload-list/upload.item.component.js

```diff
@@ -290,196 +201,146 @@ export default class UploadItemComponent extends Component {
-     * @memberof UploadItemComponent
-     */
     deleteFile() {
-        this.setState(
-            () => ({ deleted: true }),
-            () => this.props.deleteFile(this.props.adminUiConfig, this.state.struct, this.handleFileDeleted),
-        );
+        const { failed } = this.state;
+        const { item } = this.props;
+
+        if (failed) {
+            this.props.removeItemsToUpload([item]);
+            this.handleFileDeleted(item);
+        } else {
+            this.props.deleteFile(this.state.struct, this.handleFileDeleted, this.handleFileDeletedError);
+        }
     }
 
-    /**
-     * Handles the file deleted event
-     *
-     * @method handleFileDeleted
-     * @memberof UploadItemComponent
-     */
     handleFileDeleted() {
-        this.props.onAfterDelete(this.props.data);
+        this.props.onAfterDelete(this.props.item);
+        this.setState({ deleted: true });
     }
 
-    /**
-     * Returns content type identifier
-     * based on Content object returned from server after upload
-     *
-     * @method getContentTypeIdentifier
-     * @memberof UploadItemComponent
-     * @returns {String|null}
-     */
     getContentTypeIdentifier() {
-        const { contentTypesMap, data } = this.props;
+        const { contentTypesMap, item } = this.props;
 
-        if (!data.struct || !data.struct.Content) {
+        if (!item.struct?.Content) {
             return null;
         }
 
-        const contentTypeHref = data.struct.Content.ContentType._href;
+        const contentTypeHref = item.struct.Content.ContentType._href;
         const contentType = contentTypesMap ? contentTypesMap[contentTypeHref] : null;
         const contentTypeIdentifier = contentType ? contentType.identifier : null;
 
         return contentTypeIdentifier;
     }
 
-    /**
-     * Renders an icon of a content type
-     *
-     * @method renderIcon
-     * @returns {JSX.Element|null}
-     */
     renderIcon() {
+        const { failed } = this.state;
         const contentTypeIdentifier = this.getContentTypeIdentifier();
 
-        if (!contentTypeIdentifier) {
+        if (!contentTypeIdentifier || failed) {
             return null;
         }
 
-        const contentTypeIconUrl = ibexa.helpers.contentType.getContentTypeIconUrl(contentTypeIdentifier);
+        const { instanceUrl } = getRestInfo();
+        const contentTypeIconUrl = getContentTypeIconUrl(contentTypeIdentifier);
+        const [, iconName] = contentTypeIconUrl.split('#');
+        const isStandaloneMode = window.origin !== instanceUrl;
 
-        return <Icon customPath={contentTypeIconUrl} extraClasses="ibexa-icon--small-medium" />;
+        return (
+            <>
+                {isStandaloneMode ? (
+                    <Icon name={iconName} extraClasses="ibexa-icon--small" defaultIconName="file" />
+                ) : (
+                    <Icon customPath={contentTypeIconUrl} extraClasses="ibexa-icon--small" />
+                )}
+            </>
+        );
```

Comment:
```suggestion
        if (isStandaloneMode) {
            return <Icon name={iconName} extraClasses="ibexa-icon--small" defaultIconName="file" />;
        }
        
        return <Icon customPath={contentTypeIconUrl} extraClasses="ibexa-icon--small" />;
```
I would say that this version - even though it uses two returns - is cleaner to read... and has less lines. :P But your call